packages/@aws-cdk/aws-events-targets/test/sqs/integ.sqs-event-rule-target.ts

```diff
@@ -0,0 +1,39 @@
+import events = require('@aws-cdk/aws-events');
+import iam = require('@aws-cdk/aws-iam');
+import { PolicyStatementEffect } from '@aws-cdk/aws-iam';
+import sqs = require('@aws-cdk/aws-sqs');
+import cdk = require('@aws-cdk/cdk');
+import targets = require('../../lib');
+
+// ---------------------------------
+// Define a rule that triggers an event every 1min.
+// Connect the event with a queue. This means that the queue should have
+// a message sent to it every minute.
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'aws-cdk-sqs-event-target');
+
+const event = new events.Rule(stack, 'EveryMinute', {
+  scheduleExpression: 'rate(1 minute)',
+});
+
+const queue = new sqs.Queue(stack, 'MyQueue');
+const policy = new iam.PolicyStatement(PolicyStatementEffect.Allow);
```

Comment:
I was expecting this to be implicit... Why do we need this?

Reply:
This is something weird: in the beginning, I didn't put the policy explicitly. After checking it, I realized I got an error during testing:

```
IAM Statement Changes
┌───┬────────────────────────┬────────┬────────────────────────┬──────────────────────────────────┬─────────────────────────────────────────────┐
│   │ Resource               │ Effect │ Action                 │ Principal                        │ Condition                                   │
├───┼────────────────────────┼────────┼────────────────────────┼──────────────────────────────────┼─────────────────────────────────────────────┤
│ - │ ${MyQueueE6CA6235.Arn} │ Allow  │ sqs:GetQueueAttributes │ Service:events.${AWS::URLSuffix} │ "ArnEquals": {                              │
│   │                        │        │ sqs:GetQueueUrl        │                                  │   "aws:SourceArn": "${EveryMinute2BBCEA8F}" │
│   │                        │        │ sqs:SendMessage        │                                  │ }                                           │
│   │                        │        │ sqs:SendMessageBatch   │                                  │                                             │
└───┴────────────────────────┴────────┴────────────────────────┴──────────────────────────────────┴─────────────────────────────────────────────┘
(NOTE: There may be security-related changes not in this list. See http://bit.ly/cdk-2EhF7Np)
Resources
[~] AWS::SQS::QueuePolicy MyQueuePolicy6BBEDDAC
 └─ [~] PolicyDocument
     └─ [~] .Statement:
         └─ @@ -6,13 +6,6 @@
            [ ]   "sqs:GetQueueAttributes",
            [ ]   "sqs:GetQueueUrl"
            [ ] ],
            [-] "Condition": {
            [-]   "ArnEquals": {
            [-]     "aws:SourceArn": {
            [-]       "Ref": "EveryMinute2BBCEA8F"
            [-]     }
            [-]   }
            [-] },
            [ ] "Effect": "Allow",
            [ ] "Principal": {
            [ ]   "Service": {
            @@ -33,33 +26,5 @@
            [ ]         "Arn"
            [ ]       ]
            [ ]     }
            [-]   },
            [-]   {
            [-]     "Action": [
            [-]       "sqs:SendMessage",
            [-]       "sqs:SendMessageBatch",
            [-]       "sqs:GetQueueAttributes",
            [-]       "sqs:GetQueueUrl"
            [-]     ],
            [-]     "Effect": "Allow",
            [-]     "Principal": {
            [-]       "Service": {
            [-]         "Fn::Join": [
            [-]           "",
            [-]           [
            [-]             "events.",
            [-]             {
            [-]               "Ref": "AWS::URLSuffix"
            [-]             }
            [-]           ]
            [-]         ]
            [-]       }
            [-]     },
            [-]     "Resource": {
            [-]       "Fn::GetAtt": [
            [-]         "MyQueueE6CA6235",
            [-]         "Arn"
            [-]       ]
            [-]     }
            [ ]   }
            [ ] ]
Error: Some stacks have changed. To verify that they still deploy successfully, run: 'npm run integ sqs/integ.sqs-event-rule-target.js'
    at main (/Users/matteo/Github/aws-cdk/tools/cdk-integ-tools/bin/cdk-integ-assert.js:34:15)
Error: cdk-integ-assert exited with error code 1
Tests failed. Total time (12.8s) | cdk-integ-assert (7.3s) | /Users/matteo/Github/aws-cdk/tools/cdk-build-tools/node_modules/jest/bin/jest.js (5.5s)
```

And I saw that the template generated was missing the conditions specified. This is the reason I added them explicitly.

Reply:
Those permissions should *not* be in the test. If they are required, something is not working.
Could it be that you already have a stack deployed to your account and that's why the test fails on diff? Try to delete this stack and rerun without the permissions (and verify that the flow actually works).

Reply:
I solve the issue with the expected results and revert the changes over the package.json files